build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts

```diff
@@ -14,7 +14,7 @@ val localRepoElements by configurations.creating {
     }
 }
 
-val localRepoDir = layout.buildDirectory.dir("local-maven-repo")
+val localRepoDir = rootProject.layout.buildDirectory.dir("local-maven-repo")
```

Comment:
That would break multi-module concurrent execution. There's `cleanLocalRepository` to wipe the repo, and every module would run it at random, so you'll end up with half-empty `local-maven-repo`.

Reply:
Oh, you are saying that folks run `./gradlew cleanLocalRepository publishAllPublicationsToTmp-mavenRepository` in a single invocation?

I did not think of that indeed.

Let me see about figuring it out. Because the current setup really is messy for testing changes that require new versions of the whole stack.

Reply:
>Let me see about figuring it out. Because the current setup really is messy for testing changes that require new versions of the whole stack.

Could you clarify "messy"?
running with composite builds or `publishToMavenLocal` should work if you mean testing the changes for external project.

I would say the current setup is just fine for testing changes.
The only required bit is to add a PGP key and pass it to the existing test `SigstorePublishSignTest`

Reply:
Ok, I missed that the publication depends on the clean up task always.

Accidentally, the problem you describe cannot happen right now: Gradle marks all subsequent runs of the `clean` task as up-to-date. I believe this is because for Gradle the fingerprint of those tasks is a single value. But that is brittle anyway.

Looking for a better solution.

Reply:
`publishToMavenLocal` and unicorns die 😉 

I really want Gradle to get rid of that option and instead recommend usage of local repositories. That's my attempt in tweaking this.

But I see that you have a different perspective. I'll back out this change for now as it is unrelated.

Reply:
(thank you GH for not showing me the update when I replied on my before last message)

Reply:
> I really want Gradle to get rid of that option and instead recommend usage of local repositories

I do not like `publishToMavenLocal` either, however, I think it would be nice if Gradle provided something like `publishToRootProjectRepo`.